core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -94,8 +112,28 @@ private[spark] object ResourceUtils extends Logging {
 
   def parseResourceRequirements(sparkConf: SparkConf, componentName: String)
     : Seq[ResourceRequirement] = {
-    parseAllResourceRequests(sparkConf, componentName).map { request =>
-      ResourceRequirement(request.id.resourceName, request.amount)
+    listResourceIds(sparkConf, componentName).map { resourceId =>
+      val settings = sparkConf.getAllWithPrefix(resourceId.confPrefix).toMap
+      val amountDouble = settings.getOrElse(AMOUNT,
+        throw new SparkException(s"You must specify an amount for ${resourceId.resourceName}")
+      ).toDouble
```

Comment:
If we are going to allow the resource amount to be double, then we'd better update all the places where `AMOUNT` is referred to, e.g. https://github.com/apache/spark/pull/26078/files#diff-cf8a27bbb66e5c8fa3f141de053fd2d7L68

Reply:
I am adding a header to the case class `ResourceRequirement` to help clear this. `AMOUNT` only makes sense as floating point when working with task resource requests. `ResourceRequirements` are at the executor level, and it doesn't make sense to have a fractional resource there, so those should remain being integers. Perhaps `AMOUNT` is a bit too general, but I am not sure we want to change that at this point.